android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java

```diff
@@ -64,13 +64,21 @@ public void onReceive(Context context, Intent intent) {
       switch (action) {
         case PUSH_ACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnActivate, errorInfo);
-          returnMethodCallResult(resultForActivate, errorInfo);
-          resultForActivate = null;
+          if (resultForActivate != null){
+            returnMethodCallResult(resultForActivate, errorInfo);
+            resultForActivate = null;
+          }else{
+            Log.e(TAG, "resultForActivate is null: ", new NullPointerException());
+          }
           break;
         case PUSH_DEACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnDeactivate, errorInfo);
-          returnMethodCallResult(resultForDeactivate, errorInfo);
-          resultForDeactivate = null;
+          if (resultForDeactivate != null){
+            returnMethodCallResult(resultForActivate, errorInfo);
```

Comment:
I think you should use the `resultForDeactivate` in this method :wink: 
```suggestion
            returnMethodCallResult(resultForDeactivate, errorInfo);
```

Reply:
Well spotted! Can't believe I missed that on my first review pass. 😳

Reply:
Oops 😮

Reply:
Thanks for spotting this serious copy/paste error @KacperKluka . https://github.com/ably/ably-flutter/pull/303/commits/20cf0187930b52654aca95a5151e69b34b8b271c